wp-content/themes/core/components/blocks/gallery_grid/Gallery_Grid_Controller.php

```diff
@@ -0,0 +1,346 @@
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		$this->content_classes[] = 'gallery-layout--' . $this->grid_layout;
+	
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
+			return [];
+		}
+
+		return [
+			Content_Block_Controller::TAG     => 'header',
+			Content_Block_Controller::TITLE   => $this->get_title(),
+			Content_Block_Controller::CONTENT => $this->get_content(),
+			Content_Block_Controller::CLASSES => [
+				'c-block__content-block',
+				'c-block__header',
+				'b-gallery-grid'
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_title(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::TAG     => 'h2',
+			Text_Controller::CLASSES => [
+				'c-block__title',
+				'b-gallery-grid__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_slideshow_title() {
+		return $this->title ?? '';
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-gallery-grid__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_slideshow_button() {
+		return [
+			Button_Controller::CLASSES    => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after',
+				'icon-arrow-right'
+			],
+			Button_Controller::ATTRS      => [ 'data-js'  => 'dialog-trigger', 'data-content' => 'dialog-content-' . $this->id ],
+			Button_Controller::CONTENT    => esc_html__( 'View slideshow', 'tribe' ),
+		];
+	}
+
+	/**
+	 * Get the Slider
+	 *
+	 * @return array
+	 */
+	public function get_slider_args(): array {
+		$main_attrs = [];
+
+		$swiper_options = '{"preloadImages":"false","lazy":"true","watchSlidesVisibility":"true","spaceBetween":60,"ally":"true","keyboard":"true","grabCursor":"true","navigation":{"nextEl":".swiper-button-next","prevEl":".swiper-button-prev"}}';
+
+		$main_attrs['data-swiper-options'] = $swiper_options;
+
+		$slider = [
+			Slider_Controller::SLIDES          => $this->get_slides(),
+			Slider_Controller::SHOW_CAROUSEL   => false,
+			Slider_Controller::SHOW_ARROWS     => true,
+			Slider_Controller::SHOW_PAGINATION => false,
+			Slider_Controller::MAIN_ATTRS      => $main_attrs,
+			Slider_Controller::CLASSES         => [ 'b-gallery-grid__slider' ],
+		];
+
+		return $slider;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_gallery_img_ids(): array {
+		$gallery_ids = [];
```

Comment:
I think there's probably a way to simplify this method a bit, but I'll leave it to a BE to suggest any changes. ;)

Reply:
return ! empty( $this->gallery ) ? array_filter( wp_list_pluck( $this->gallery, 'id' ) ) : [];